### PR TITLE
#150512346 - make recipe favorite 

### PR DESCRIPTION
**Pull request by llabake**
#### What does this PR do?
Implement endpoint to get user's favorite recipe
#### Description of Task to be completed?
Get favorite recipes from (`/api/users/<userId>/recipes`)
#### What are the relevant pivotal tracker stories?
#150512346
#### Screenshots
image

### CODE REVIEW

File: server/controllers/v1/recipeController.js [hunk 1]
```diff
@@ -85,3 +85,34 @@ export const voteRecipe = (req, res) => {
     }
   });
 };
+
+export const addRecipeAsFavorite = (req, res) => {
```
Comment (nelsonabienorin) [inline server/controllers/v1/recipeController.js:89]: ES 5 codes are not allowed

File: server/migrations/20170905224636-create-favorite-recipe.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+module.exports = {
```
Comment (nelsonabienorin) [inline server/migrations/20170905224636-create-favorite-recipe.js:1]: ES 5 codes are not allowed

File: server/migrations/20170905225209-favorite-recipeId-userId-unique.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
```
Comment (nelsonabienorin) [inline server/migrations/20170905225209-favorite-recipeId-userId-unique.js:1]: ES 5 codes are not allowed

File: server/models/favoriterecipe.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      allowNull: false
+    },
+    recipeId: {
+      type: DataTypes.INTEGER,
+      allowNull: false
+    }
+  }, {
+    classMethods: {
+      associate() {
+        // associations can be defined here
```
Comment (nelsonabienorin) [inline server/models/favoriterecipe.js]: I would love to see the associations defined here

File: server/routes/recipeRoute.js [hunk 5]
```diff
@@ -13,6 +13,10 @@ const recipeRoute = (app) => {
     recipeController.getAllRecipes);
   app.post('/api/v1/recipes/:recipeId(\\d+)/votes/', authMiddleware, recipeExist,
     recipeController.voteRecipe);
+  app.post('/api/v1/users/:userId(\\d+)/recipes/:recipeId(\\d+)/', authMiddleware, recipeExist,
```
Comment (nelsonabienorin) [inline server/routes/recipeRoute.js:16]: why do you have to use this (\\d+) in your API?

File: server/controllers/v1/recipeController.js [hunk 6]
```diff
@@ -85,3 +85,34 @@ export const voteRecipe = (req, res) => {
...
+      message: 'Recipe has been favorited' }))
+    .catch((error) => {
+      if (error.name === 'SequelizeUniqueConstraintError') {
+        res.status(400).json({ message: 'Recipe has already been favorited' });
+      } else {
+        res.status(400).json({ error });
+      }
+    });
+};
+
```
Comment (nelsonabienorin) [inline server/controllers/v1/recipeController.js:104]: I will like to see comments in your codebase, you are not adding comments where necessary

**Review by nelsonabienorin — CHANGES_REQUESTED** [review]
Good job! Kindly address my comments and make changes where necessary.

File: server/routes/recipeRoute.js (hunk 5 above)
Comment (llabake) [reply server/routes/recipeRoute.js:16]: This is to check that its only numeric data type that is passed in.

File: server/models/favoriterecipe.js (hunk 4 above)
Comment (llabake) [reply server/models/favoriterecipe.js]: Ok. writing the association

File: server/migrations/20170905225209-favorite-recipeId-userId-unique.js (hunk 3 above)
Comment (llabake) [reply server/migrations/20170905225209-favorite-recipeId-userId-unique.js:1]: correcting all my models to es6

**Pull request merged**